### ISSUE TITLE
When `in` argument matches `in` parameter we forget to record the match in escape analysis

### ISSUE THREAD

**Issue by VSadov** [dotnet/roslyn#22456]
As a result we may report false and misleading escape errors:

```cs
        static ref readonly int F1(in int x)
        {
            return ref F1(in x); // false eror here
        }
```
Since we did not mark "x" parameter as matched by an argument, we think it was supplied an optional value and therefore the whole call is not safe to return by reference.
